app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+class CommentsController < ApplicationController
+  before_action :set_commentable
+
+  def create
+    @comment = @commentable.comments.new(comment_params)
+    @comment.user = current_user
+    if @comment.save
+      set_comment_variables
+      respond_with_format(@commentable)
+    else
+      post_redirect_path(@commentable)
+    end
+  end
+
+  private
+
+  def set_commentable
+    @params = params[:comment]
+    id = @params[:commentable_id]
+    @commentable = if @params[:commentable_type] == 'Post'
+                     Post.find(id)
+                   else
+                     Comment.find(id)
+                   end
+  end
+
+  def comment_params
+    params.require(:comment).permit(:content, :commentable_type, :commentable_id)
+  end
+
+  def post_redirect_path(commentable)
+    if commentable.is_a?(Post)
+      post_path(commentable)
+    else
+      post_redirect_path(commentable.commentable)
+    end
+  end
+
+  def set_comment_variables
+    is_post = @commentable.is_a?(Post)
+    @comment_count = @commentable.comments.count if is_post
+  end
```

Comment:
I feel like this method might not be necessary.

What if you just call `@commentable&.comments&.count` from the view?